packages/internals/src/engine-commands/formatSchema.ts

```diff
@@ -1,52 +1,126 @@
 import Debug from '@prisma/debug'
-import { BinaryType } from '@prisma/fetch-engine'
-import execa from 'execa'
 import fs from 'fs'
+import { match } from 'ts-pattern'
 
-import { resolveBinary } from '../resolveBinary'
+import { ErrorArea, RustPanic } from '../panic'
+import { prismaFmt } from '../wasm'
 
 const debug = Debug('prisma:formatSchema')
 
-const MAX_BUFFER = 1_000_000_000
+type FormatSchemaParams = { schema: string; schemaPath?: never } | { schema?: never; schemaPath: string }
 
-// can be used by passing either
-// the schema as a string
-// or a path to the schema file
-export async function formatSchema({ schema }: { schema: string })
-export async function formatSchema({ schemaPath }: { schemaPath: string })
-export async function formatSchema({ schemaPath, schema }: { schemaPath?: string; schema?: string }): Promise<string> {
+function isSchemaOnly(schemaParams: FormatSchemaParams): schemaParams is { schema: string; schemaPath?: never } {
+  return !!schemaParams.schema
```

Comment:
```suggestion
  return Boolean(schemaParams.schema)
```
Nit: It's more readable